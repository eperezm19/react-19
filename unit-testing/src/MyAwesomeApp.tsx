const firstName = "John";
const lastName = "Doe";
const favoriteGames = [
  "The Witcher 3",
  "The Last of Us",
  "The Elder Scrolls V: Skyrim",
];

export const MyAwesomeApp = () => {
  return (
    <>
      <h1>{firstName}</h1>
      <h2>{lastName}</h2>

      <p>{favoriteGames.join(", ")}</p>
      <p>{2 + 2}</p>
    </>
  );
};
