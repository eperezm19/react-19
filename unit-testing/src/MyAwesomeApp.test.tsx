import { test, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyAwesomeApp } from "./MyAwesomeApp";
describe("MyAwesomeApp", () => {
  test("should render firstName and lastName", () => {
    const { container } = render(<MyAwesomeApp />);

    const h1 = container.querySelector("h1");
    expect(h1?.innerHTML).toBe("John");
    const h2 = container.querySelector("h2");
    expect(h2?.innerHTML).toBe("Doe");
  });
  test("should render firstName and lastName --screen", () => {
    render(<MyAwesomeApp />);

    const h1 = screen.getByRole("heading", { level: 1 });
    expect(h1?.innerHTML).toBe("John");
    const h2 = screen.getByRole("heading", { level: 2 });
    expect(h2?.innerHTML).toBe("Doe");
  });

  test(' should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });
});
