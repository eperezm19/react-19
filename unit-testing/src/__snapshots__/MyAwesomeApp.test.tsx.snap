// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp >  should match snapshot 1`] = `
<div>
  <h1>
    John
  </h1>
  <h2>
    Doe
  </h2>
  <p>
    The Witcher 3, The Last of Us, The Elder Scrolls V: Skyrim
  </p>
  <p>
    4
  </p>
</div>
`;
